src/odemis/util/comp.py

```diff
@@ -107,4 +109,57 @@ def get_fov_rect(comp, fov):
     return (center[0] - fov[0] / 2,  # left
             center[1] - fov[1] / 2,  # top
             center[0] + fov[0] / 2,  # right
-            center[1] + fov[1] / 2)  # bottom
\ No newline at end of file
+            center[1] + fov[1] / 2)  # bottom
+
+
+MAX_ZLEVELS = 500
+
+def generate_zlevels(focuser, zrange, zstep):
+    """
+    Calculates the zlevels for a zstack acquisition, using the zmax, zmin
+    and zstep, as well as the current focus z position
+    focuser (Actuator): focus component
+    zrange (float, float): contains the zMin and zMax, respectively. Error is raised
+        if the order is not (zMin, zMax). 
+    zstep (float>0 or float<0): distance between two successive zlevels. If negative,
+        the order of the returned list of zlevels is reversed. The actual zstep in the 
+        returned list is adjusted to the closest value of the given zstep that divides
+        the zrange. If zstep is too small, this will lead to too large number of zlevels. 
+        In that case IndexError is raised. 
+    returns (list of floats): list of zlevels, where each zlevel is absolute position for
+        the focuser.  
+    """
+    if zstep == 0:
+        raise ZeroDivisionError("The step size 'Zstep' can not be zero")
+    if zrange[0] > zrange[1]:
+        raise ValueError("The given range is not correct. zmin should be first and zmax second")
+    if "z" not in focuser.axes.keys():
+        raise KeyError("The focus actuator %s does not have z axis" %focuser)
+
+    focuser_pos = focuser.position.value["z"]
+    focuser_rng = focuser.axes["z"].range
+    # clip the zMax and zMin to the actuator limits if necessary 
+    if zrange[1] + focuser_pos > focuser_rng[1]:
+        zrange[1] = focuser_rng[1] - focuser_pos
+    if focuser_pos + zrange[0] < focuser_rng[0]:
+        zrange[0] = focuser_rng[0] - focuser_pos
+    if zrange[0] == zrange[1]:
+        return focuser_pos + zrange[0]
+
+    # find number of samples 
+    n = (zrange[1] - zrange[0]) / abs(zstep) + 1
+    if n > MAX_ZLEVELS:
+        raise IndexError("The number of zlevels is too large. Reduce the zstep value.")
```

Comment:
Please indicate in the error message: the number of zlevels, and the maximum number allowed. In case a user triggers this error, it'll be much easier to find out why, and how to fix it.